File: lab-2/src/moves/Facade.java
package moves;

import ru.ifmo.se.pokemon.*;

public class Facade extends PhysicalMove {
    public Facade() {
        super(Type.NORMAL, 70, 100);
    }

    protected void applyOppDamage(Pokemon def, double damage) {
        Status defStatus = def.getCondition();
        if (defStatus.equals(Status.BURN)
                || defStatus.equals(Status.POISON)
                || defStatus.equals(Status.PARALYZE)) {
            super.applyOppDamage(def, damage*2);
        }
        else {
            super.applyOppDamage(def, damage);
        }
    }

    protected String describe() {
        return "is using Facade";
    }
}

File: lab-2-kotlin/src/moves/Facade.kt
package moves

import ru.ifmo.se.pokemon.*

class Facade : PhysicalMove(Type.NORMAL, 70.0, 100.0) {
    override fun applyOppDamage(def: Pokemon, damage: Double) {
        val defStatus = def.condition
        if (defStatus == Status.BURN
                || defStatus == Status.POISON
                || defStatus == Status.PARALYZE) {
            super.applyOppDamage(def, damage * 2)
        } else {
            super.applyOppDamage(def, damage)
        }
    }

    override fun describe(): String {
        return "is using Facade"
    }
}